api/src/main/java/org/pmiops/workbench/api/CohortAnnotationDefinitionController.java

```diff
@@ -100,7 +100,24 @@ public CohortAnnotationDefinition apply(org.pmiops.workbench.db.model.CohortAnno
                                      Long cohortId,
                                      Long annotationDefinitionId,
                                      ModifyCohortAnnotationDefinitionRequest modifyCohortAnnotationDefinitionRequest) {
-        return ResponseEntity.status(HttpStatus.NOT_IMPLEMENTED).body(new CohortAnnotationDefinition());
+        Cohort cohort = findCohort(cohortId);
+        //this validates that the user is in the proper workspace
+        validateMatchingWorkspace(workspaceNamespace, workspaceId, cohort.getWorkspaceId());
+
+        org.pmiops.workbench.db.model.CohortAnnotationDefinition cohortAnnotationDefinition =
+                cohortAnnotationDefinitionDao.findOne(annotationDefinitionId);
+
+        if (cohortAnnotationDefinition == null) {
+            throw new BadRequestException(
+                    String.format("Invalid Request: No Cohort Annotation Definition exists for annotationDefinitionId: %s",
+                            annotationDefinitionId));
+        }
+        cohortAnnotationDefinition.columnName(modifyCohortAnnotationDefinitionRequest.getName());
```

Comment:
Nit: use setColumnName? (I find that a bit clearer.)

Since we're going to reference them by name, we should enforce that annotation definition names are unique within cohorts... probably both by adding a check here (and in createCohortAnnotationDefinition) and by making a unique index in the database.

You may want to validate that the cohort ID matches the cohort ID parameter above as well.

Reply:
changed to columnName, added unique contraints on column_name and added cohortId as param on dao call to findByCohortIdAndCohortAnnotationDefinitionId.

Reply:
Also, changed BadRequestExceptions to NotFoundExceptions. Seems like the appropriate thing.